### PR TITLE
Turning witness columns into constants.

### PR DESCRIPTION
**Pull request by lvella**

### CODE REVIEW

**Review by onurinanc — COMMENTED** [review]
You can also do the same for `do_mload`

**Comment by lvella** [comment]
> You can also do the same for `do_mload`

I tried, exceeds the maximum polynomial degree or something.

**Review by onurinanc — APPROVED** [review]

**Review by georgwiese — APPROVED** [review]
LGTM, although I think the expression can be simplified further.

File: std/machines/hash/poseidon_bb.asm [hunk 1]
```diff
@@ -157,10 +154,7 @@ machine PoseidonBB(mem: Memory, split_bb: SplitBB) with
     link if do_mstore ~> mem.mstore(high_addr, low_addr, time_step + 1, word_high, word_low);
 
     // Make sure that in row i + STATE_SIZE, word_low and word_high correspond to output i
-    ///////////////////// DEBUG: current_output as a witness. ///////////////////////
-    let current_output;
-    /////////////////////////////////////////////////////////////////////////
-    current_output = array::sum(array::new(OUTPUT_SIZE, |i| CLK[i + STATE_SIZE] * output[i]));
+    let current_output = array::sum(array::new(OUTPUT_SIZE, |i| CLK[i + STATE_SIZE] * output[i]));
```
[the remark this answers is not in the document]
Comment (lvella) [reply std/machines/hash/poseidon_bb.asm:157]: I don't know, this was copied from poseidon_gl. Let me try.

**Review by georgwiese — APPROVED** [review]

**Review by onurinanc — APPROVED** [review]
LGTM

**Pull request merged**